test/clj/rems/browser_test_util.clj

```diff
@@ -299,3 +308,46 @@
        (remove nil?)
        (mapv value-of-el)
        first))
+
+(defn check-axe
+  "Runs automated accessibility tests using axe.
+
+  Returns the test report.
+
+  See https://www.deque.com/axe/"
+  []
+  (let [result (js-async "
+    var args = arguments;
+    var callback = args[args.length-1];
+    window.axe.configure({'reporter': 'v2'});
+    window.axe.run({ exclude:[['.dev-reload-button'],
+                              ['#figwheel-heads-up-container']]})
+    .then(callback);")]
+    result))
+
+(defn gather-axe-results
+  "Runs automatic accessbility tests using `check-axe`
+  and appends them to the test context for summary reporting."
+  []
+  (context-update! :axe conj-vec (check-axe)))
+
+(defn accessibility-report-fixture
+  "Runs the tests and finally stores the gathered accessibility
+  results into summary files.
+
+  NB: the individual tests must call `gather-axe-results` in each
+  interesting spot to have a sensible report to write."
+  [f]
+  (try
+    (f)
+    (finally
+      (doseq [k (->> (context-get :axe)
```

Comment:
so are the keys like `violations`, `warnings` etc.? this code feels a bit too cryptic for that

Reply:
Yes the JSON has plenty of keys like `violations`, `passes`, `incomplete` and `inapplicable`. It doesn't make sense to list them in the code. On the other hand, at the beginning we probably have to have a look at what the different files contain from time to time to understand the reports better before we can for example to say to only care about the `violations`.

Reply:
I prefer the `violations` or whatever we end up with to be a simple file with not much extra. Therefore I don't want to just dump the whole thing to one file. Also I think we want to merge the results of different checks to one file.

Reply:
Fair enough.